demos/jobs/jobs_demo_mosquitto/jobs_demo_mosquitto.c

```diff
@@ -0,0 +1,1264 @@
+
+    assert( h != NULL );
+    assert( message != NULL );
+    assert( ( message->payload != NULL ) && ( message->payloadlen > 0 ) );
+
+    json_ret = JSON_Validate( message->payload, message->payloadlen );
+
+    if( json_ret != JSONSuccess )
+    {
+        warnx( "invalid job document" );
+    }
+    else
+    {
+        json_ret = JSON_Search( message->payload,
+                                message->payloadlen,
+                                "execution.jobId",
+                                ( sizeof( "execution.jobId" ) - 1 ),
+                                '.',
+                                &jobid,
+                                &jobidLength );
+    }
+
+    if( json_ret == JSONSuccess )
+    {
+        json_ret = JSON_Search( message->payload,
+                                message->payloadlen,
+                                "execution.jobDocument.url",
+                                ( sizeof( "execution.jobDocument.url" ) - 1 ),
+                                '.',
+                                &url,
+                                &urlLength );
+    }
+
+    if( json_ret == JSONSuccess )
+    {
+        jobid = strndup( jobid, jobidLength );
+        url = strndup( url, urlLength );
+        assert( ( jobid != NULL ) && ( url != NULL ) );
+
+        h->jobid = jobid;
+        h->jobidLength = jobidLength;
+        h->url = url;
+        h->urlLength = urlLength;
+        ret = true;
+    }
+    else
+    {
+        if( jobid != NULL )
+        {
+            jobid[ jobidLength ] = '\0';
+            warnx( "missing url; failing job id: %s", jobid );
+            ( void ) sendUpdate( h, jobid, jobidLength, makeReport_( "FAILED" ) );
+        }
+    }
+
+    return ret;
+}
+
+/*-----------------------------------------------------------*/
+
+void on_message( struct mosquitto * m,
+                 void * p,
+                 const struct mosquitto_message * message )
+{
+    handle_t * h = p;
+    JobsStatus_t ret;
+    JobsTopic_t api;
+    char * jobid;
+    uint16_t jobidLength;
+
+    assert( h != NULL );
+    assert( message->topic != NULL );
+
+    /* identify a Jobs API and a job ID represented in a topic buffer */
+    ret = Jobs_MatchTopic( message->topic,
+                           strlen( message->topic ),
+                           h->name,
+                           h->nameLength,
+                           &api,
+                           &jobid,
+                           &jobidLength );
+    assert( ret != JobsBadParameter );
+    ( void ) ret;
+
+    switch( api )
+    {
+        /* a job has been added or the current job was canceled */
+        case JobsNextJobChanged:
+            h->lastPrompt = 0;
+            h->lastUpdate = 0;
+            break;
+
+        /* response to a request to start the next job */
+        case JobsStartNextSuccess:
+
+            if( h->runStatus == None )
+            {
+                if( parseJob( h, message ) == true )
```

Comment:
nit
```suggestion
                if( h->runStatus == None && parseJob( h, message ) == true )
```

Reply:
That doesn't have the same meaning.  In the original, the else case is independent of the outcome of parseJob().